subsys/shell/backends/shell_telnet.c

```diff
@@ -166,15 +166,25 @@ static inline bool telnet_handle_command(struct net_pkt *pkt)
 	cmd = (struct telnet_simple_command *)net_pkt_get_data(pkt,
 							       &cmd_access);
 	if (!cmd || cmd->iac != NVT_CMD_IAC) {
-		return false;
+		return 0;
 	}
 
 	if (IS_ENABLED(CONFIG_SHELL_TELNET_SUPPORT_COMMAND)) {
 		LOG_DBG("Got a command %u/%u/%u", cmd->iac, cmd->op, cmd->opt);
 		telnet_reply_command(cmd);
 	}
 
-	return true;
+	if (cmd->op == NVT_CMD_SB) {
+		/* TODO Add subnegotiation support. */
+		return -EOPNOTSUPP;
+	}
+
+	if (cmd->op == NVT_CMD_WILL || cmd->op == NVT_CMD_WONT ||
+	    cmd->op == NVT_CMD_DO || cmd->op == NVT_CMD_DONT) {
+		return 3;
```

Comment:
Is it possible to replace these magic numbers (for me) with some defined error codes?

Reply:
I've replaced the numbers with appropriate symbols.